python/contrib/dag/__init__.py

```diff
@@ -69,6 +85,10 @@ def add_edge(self, ind_node, dep_node, graph=None):
         """ Add an edge (dependency) between the specified nodes. """
         if not graph:
             graph = self.graph
+
+        # Allow original graph to grow
+        self.__original_graph = None
```

Comment:
Ah, I see what you are doing... I think a better name/method would be to call this thing `__cached_graph`. Then when you add new nodes and edges you "invalidate the cache" or set it to `None`.